### PR TITLE
Feature/4 manage books

### PR DESCRIPTION
**Pull request by LukaM1999**
- infinite scrolling with tanstack query and intersection observer
- book search by title with debounce
- axios config relocation to dedicated file

### CODE REVIEW

File: library-front/src/components/BookCard/BookCard.tsx [hunk 1]
```diff
@@ -1,8 +1,72 @@
-import React, { FC } from 'react'
+import { FC } from 'react'
+import { AiOutlineEdit } from 'react-icons/ai'
```
Comment (aleksandarkrnjaic) [inline library-front/src/components/BookCard/BookCard.tsx:2]: You can alias this so it's more readable. eg  as EditIcon

File: library-front/src/components/BookCard/BookCard.tsx [hunk 2]
```diff
@@ -1,8 +1,72 @@
...
+import { FiDelete } from 'react-icons/fi'
```
Comment (aleksandarkrnjaic) [inline library-front/src/components/BookCard/BookCard.tsx:3]: same

File: library-front/src/components/Header/Header.tsx [hunk 3]
```diff
@@ -1,26 +1,50 @@
...
-const Header: FC<HeaderProps> = ({ jwt, setJwt }) => {
+const Header: FC<HeaderProps> = ({ jwt, setJwt, setSearch }) => {
+  const [isHeaderVisible, setIsHeaderVisible] = useState(true)
+  const [prevScrollPos, setPrevScrollPos] = useState(0)
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const currentScrollPos = window.pageYOffset
+
+      setIsHeaderVisible(prevScrollPos > currentScrollPos || currentScrollPos < 50)
```
Comment (aleksandarkrnjaic) [inline library-front/src/components/Header/Header.tsx:21]: does this mean we have to scroll way back to the top to have visible header?

**Review by aleksandarkrnjaic — APPROVED** [review]

File: library-front/src/components/Header/Header.tsx (hunk 3 above)
Comment (LukaM1999) [reply library-front/src/components/Header/Header.tsx:21]: No, the prevScrollPos > currentScrollPos means that when the user starts scrolling up, the header becomes visible

**Pull request merged**